ex2/ex2.c

```diff
@@ -2,13 +2,55 @@
 // and then calls `fork()` to create a new process. Can both the child and parent access the file descriptor 
 // returned by `fopen()`? What happens when they are written to the file concurrently?
 
+// both the parent and the child cannot access the file descriptor returned by fopen()
+// when they are written to the file concurrently the parent process takes precedence over the child process
+
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
 
 int main(int argc, char* argv[])
 {
     // Your code here 
+
+    // open text.txt file using fopen()
+	FILE *fp;
+	int c;
+
+	printf("opening file text.txt\n");
+
+	fp = fopen("text.txt", "r");
+
+    // calls fork() to create a new process
+
+    int rc = fork();
+
+    if (rc < 0) {
+    	printf("fork failed\n");
+    	exit(1);
+    } else if (rc == 0) {
+    	printf("CHILD EXECUTING:\n");
+    	while(1) {
```

Comment:
You made this way more complicated than it needed to be. The child process simply needs to wrote a static string to the text file. No need to fetch input from the user.